lib/iotaUtils.js

```diff
@@ -154,6 +154,8 @@ function mergeDeviceWithConfiguration(deviceData, configuration, callback) {
             deviceData[fields[i]] = configuration[confField];
         } else if (!deviceData[fields[i]] && (!configuration || !configuration[confField])) {
             deviceData[fields[i]] = defaults[i];
+        } else {
+            config.getLogger().error(context, 'There is no possible merge');
```

Comment:
Not sure if this is ending in this branch of the multi-if is an error condition. Should be double-checked.

Reply:
Is we cannot confirm this is an error condition (can we?) then we shouldn't log it as error. What do you think?

Reply:
Hi @fgalan . I think it is not an error condition, it is only a recommendation based on:

- MISRA C:

    MISRA C:2004, 14.10 - All if...else if constructs shall be terminated with an else clause.
    MISRA C++:2008, 6-4-2 - All if...else if constructs shall be terminated with an else clause.
    MISRA C:2012, 15.7 - All if...else if constructs shall be terminated with an else statement

- CERT (The CERT C Secure Coding Standard):
    CERT, MSC01-C. - Strive for logical completeness (https://wiki.sei.cmu.edu/confluence/display/c/MSC01-C.+Strive+for+logical+completeness)
    CERT, MSC57-J. - Strive for logical completeness (https://wiki.sei.cmu.edu/confluence/display/java/MSC57-J.+Strive+for+logical+completeness)

Reply:
If this is not an error condiction, then we shouldn't use error() for logging. I'd suggest using debug(), along with a more descriptive error message. For instance:

```suggestion
            config.getLogger().debug(context, 'at field %d configuration merging logic did not merge anything', i);
```

(Not sure if %d is the right format string for integers in JavaScript... please check & test)

Reply:
Fixed. Not sured in which commit, but at least in 9dfbdcb seems to be fine.